vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabSheetTest.java

```diff
@@ -65,6 +65,23 @@ public void addTab_contentAdded() {
         Assert.assertTrue(content.getParent().isPresent());
     }
 
+    @Test
+    public void addSameTabAgainWithNewContent_oldContentRemoved() {
+        // Add a tab with content
+        var content0 = new Span("Content 0");
+        var tab0 = tabSheet.add("Tab 0", content0);
+
+        // Assert that the content is attached to the parent (the tab is
+        // selected)
+        Assert.assertEquals(tabSheet, content0.getParent().get());
+
+        // Add the same Tab instance again but with a new content component
+        tabSheet.add(tab0, new Span("Content 0"));
+
+        // Check that the old content is no longer attached to the parent
+        Assert.assertFalse(content0.getParent().isPresent());
```

Comment:
Just asking: would a developer expect that the "newly" added tab/content gets added last or at the position of the "old tab"? 

(I'm not sure myself as well, would be good to test the index tho)

Reply:
Added the test case.